components/camel-dropbox/src/main/java/org/apache/camel/component/dropbox/core/DropboxAPIFacade.java

```diff
@@ -257,58 +235,65 @@ public DropboxResult move(String remotePath, String newRemotePath) throws Dropbo
      * @throws DropboxException
      */
-    public DropboxResult get(String remotePath) throws DropboxException {
-        DropboxResult result = new DropboxFileDownloadResult();
-        //a map representing for each path the result of the baos
-        Map<String, ByteArrayOutputStream> resultEntries = new HashMap<String, ByteArrayOutputStream>();
-        //iterate from the remotePath
-        downloadFilesInFolder(remotePath, resultEntries);
-        //put the map of baos as result
-        result.setResultEntries(resultEntries);
-        return result;
+    public DropboxFileDownloadResult get(String remotePath) throws DropboxException {
+        return new DropboxFileDownloadResult(downloadFilesInFolder(remotePath));
     }
 
-    private void downloadFilesInFolder(String path, Map<String, ByteArrayOutputStream> resultEntries) throws DropboxException {
-        DbxEntry.WithChildren listing = null;
+
+    public boolean isDirectory(String path) throws DropboxException {
         try {
-            listing = DropboxAPIFacade.client.getMetadataWithChildren(path);
+            DbxEntry.WithChildren listing = client.getMetadataWithChildren(path);
+            return listing.children != null;
         } catch (DbxException e) {
             throw new DropboxException(path + " does not exist or can't obtain metadata");
         }
-        if (listing.children == null) {
-            LOG.info("downloading a single file...");
-            downloadSingleFile(path, resultEntries);
-            return;
-        }
-        for (DbxEntry entry : listing.children) {
-            if (entry.isFile()) {
-                try {
-                    //get the baos of the file
-                    downloadSingleFile(entry.path, resultEntries);
-                } catch (DropboxException e) {
-                    LOG.warn("can't download from " + entry.path);
+    }
+
+
+    private Map<String, Object> downloadFilesInFolder(String path) throws DropboxException {
+        try {
+            DbxEntry.WithChildren listing = client.getMetadataWithChildren(path);
+            if (listing == null) {
+                return Collections.emptyMap();
+            } else if (listing.children == null) {
+                LOG.info("downloading a single file...");
+                Map.Entry<String, Object> entry = downloadSingleFile(path);
+                return Collections.singletonMap(entry.getKey(), entry.getValue());
+            }
+            Map<String, Object> result = new HashMap<>();
+            for (DbxEntry entry : listing.children) {
+                if (entry.isFile()) {
+                    try {
+                        Map.Entry<String, Object> singleFile = downloadSingleFile(entry.path);
+                        result.put(singleFile.getKey(), singleFile.getValue());
+                    } catch (DropboxException e) {
+                        LOG.warn("can't download from " + entry.path);
```

Comment:
LOG.warn("Cannot download from " + entry.path + " due " + e.getMessage() + ". This exception is ignored.");